lookup/src/plookup/quotient_poly.rs

```diff
@@ -0,0 +1,263 @@
+use std::marker::PhantomData;
+
+use ark_ff::{PrimeField, Zero};
+use ark_poly::{DenseUVPolynomial, EvaluationDomain};
+
+use crate::plookup::types::PlookupEvaluations;
+use crate::poly::ExtraDensePoly;
+use crate::types::{Domain, Poly};
+
+pub struct QuotientPoly<F>(PhantomData<F>);
+
+#[allow(clippy::too_many_arguments)]
+impl<F: PrimeField> QuotientPoly<F> {
+    /// Computes quotient polynomial in `Plookup`.
+    ///
+    /// # Arguments
+    ///
+    /// * `f`
+    /// * `t`
+    /// * `h1`
+    /// * `h2`
+    /// * `z`
+    /// * `domain`
+    /// * `beta`
+    /// * `gamma`
+    ///
+    /// returns: The quotient polynomial.
+    pub fn compute_quotient_poly(
+        f: &Poly<F>,
+        t: &Poly<F>,
+        h1: &Poly<F>,
+        h2: &Poly<F>,
+        z: &Poly<F>,
+        domain: &Domain<F>,
+        beta: F,
+        gamma: F,
+    ) -> Poly<F> {
+        let domain_2n = &Domain::<F>::new(2 * domain.size()).unwrap();
+        let check_z1_zn = Self::check_z1_zn_equal_one(z, domain, domain_2n);
+        let check_z = Self::check_z_poly(f, t, h1, h2, z, gamma, beta, domain, domain_2n);
```

Comment:
If the function's name is `check_sth`, it should return a boolean or Result<> result, instead of a polynomial. 
You should change the names of them or separate those functions.